server/src/main/java/org/elasticsearch/search/aggregations/timeseries/TimeSeriesIndexSearcher.java

```diff
@@ -65,6 +67,7 @@ public void search(Query query, BucketCollector bucketCollector) throws IOExcept
         int seen = 0;
         query = searcher.rewrite(query);
         Weight weight = searcher.createWeight(query, bucketCollector.scoreMode(), 1);
+        AtomicInteger tsidOrd = new AtomicInteger();
```

Comment:
I don't think this has to be an AtomicInteger, because there is one thread here that does the time series search, right? It is just convenient to use AtomicInteger?

Reply:
Just scanning, it looks like it can be an `int`. But I might be missing something. FWIW I prefer `new int[1]` if I need a mutable int box because it doesn't make the reader think "oh god, there are multiple threads now"

Reply:
And yes it can be `int` and not `long` of course because we're dealing with single indexes here.

Reply:
uh, sorry, does it need `long` because it's ordinals across all leaves? it's not likely to get that big, but global ordinals a `long` for paranoia.

I don't think the locking will slow things down. it's uncontended so it'll get zapped. I think. I just think it's easier to read as an `int` because I never have to wonder where the threads are. Could you do `() -> tsidOrd` as the supplier and keep it as just `int`?

Reply:
Even across all leaves `maxDoc` is an int so ordinals will only ever be `int` as well.  A docvalues ordinal would be a `long` because you could have multiple values per doc, but we will only ever have at most one tsid per document.

We can't do `() -> tsidOrd` because it needs to be 'effectively final' so it would have to be an array reference.  Which is fine.

Reply:
:+1: on `int` - we know that we have at most one tsid per doc.

Array reference is better for me. It's more code but I can read it faster....

Reply:
I also lean toward using `new int[1]` here over `AtomicInteger`.